sheepdog/utils/__init__.py

```diff
@@ -534,18 +530,19 @@ def lookup_program(psql_driver, program):
 
 def proxy_request(project_id, uuid, data, args, headers, method, action,
                   indexd_client, dry_run=False):
+    """Proxy a request to an object storage"""
+
     node = get_node(project_id, uuid)
     indexd_doc = indexd_client.get(uuid)
-
-    program, project = project_id.split('-', 1)
-    s3_url = generate_s3_url(
-        host=flask.current_app.config['SUBMISSION']['host'],
-        bucket=flask.current_app.config['SUBMISSION']['bucket'],
-        program=program,
-        project=project,
-        uuid=uuid,
-        file_name=indexd_doc.file_name,
-    )
+    s3_url = None
+    key_name = None
```

Comment:
Do we need to handle the potential `None`'s here?

Reply:
If the whole url is incorrect then the check directly below it will fail.

https://github.com/uc-cdis/sheepdog/blob/fix/dont-assume-s3-url-on-delete/sheepdog/utils/__init__.py#L547

Reply:
yeah, you are right.